brainiak/searchlight/searchlight.py

```diff
@@ -0,0 +1,353 @@
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#               http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from mpi4py import MPI
+import numpy as np
+from sklearn.base import TransformerMixin
+
+"""Distributed searchlight
+
+Given a function and a list of volumes, this applies the function
+to all voxels.
+"""
+
+# Author: Michael Anderson
+
+__all__ = [
+    "Searchlight",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class Searchlight(TransformerMixin):
+    """ Class for searchlight.
+
+    A searchlight is a computation that is applied in a sliding
+    window to subsets of voxels across a volume. The radius of the
+    searchlight specifies how many voxels are involved in each
+    computation. A user-provided function is applied to all voxels
+    within the radius, provided that the mask was non-zero at that
+    point in the volume. The ouput of the searchlight analysis is
+    another volume which contains the user-provided function's
+    output at each point.
+
+    Parameters
+    ----------
+
+    rad: Positive integer indicating the radius of the searchlight
+         cube.
+
+    fn: A user-provided function which is applied to the data in a
+    searchlight.
+        This user-specified function must accept 2 parameters:
+            data: A list of 4D numpy.ndarray objects which has the same
+            structure as the 'data' parameter to the searchlight, but
+            only includes voxels within a 'rad' radius from the current
+            searchlight center
+
+            mask: A 3D numpy.ndarray object with type np.bool indicating
+            which includes voxels within a 'rad' radius from the current
+            searchlight center
+
+    Attributes
+    ----------
+    None
+    """
+
+    def __init__(self, rad, fn):
+        """ Constructor for searchlight.
+
+        rad: Positive integer indicating the radius of the searchlight
+                cube.
+
+        fn: A user-provided function which is applied to the data in a
+        searchlight.
+            This user-specified function must accept 2 parameters:
+                data: A list of 4D numpy.ndarray objects which has the same
+                structure as the 'data' parameter to the searchlight, but
+                only includes voxels within a 'rad' radius from the current
+                searchlight center
+
+                mask: A 3D numpy.ndarray object with type np.bool indicating
+                which includes voxels within a 'rad' radius from the current
+                searchlight center
```

Comment:
See comment above about function output. Please, add the description here as well.